src/model/model.ts

```diff
@@ -240,26 +257,45 @@ export class Model {
       }
     }
 
-    ast.policy = ast.policy.concat(rules);
+    const priorityFlag = ast.tokens.indexOf('p_priority') !== -1;
+
+    if (priorityFlag) {
+      rules.forEach((rule) => {
+        this.addPolicy(sec, ptype, rule);
+      });
+    } else {
+      ast.policy = ast.policy.concat(rules);
+    }
 
     return [true, rules];
   }
 
   // updatePolicy updates a policy from the model
   public updatePolicy(sec: string, ptype: string, oldRule: string[], newRule: string[]): boolean {
-    if (this.hasPolicy(sec, ptype, oldRule)) {
-      const ast = this.model.get(sec)?.get(ptype);
-      if (!ast) {
-        return false;
-      }
-      // const index = ast.policy.indexOf(oldRule);
-      const index = ast.policy.findIndex((r) => util.arrayEquals(r, oldRule));
-      if (index !== -1) {
+    const ast = this.model.get(sec)?.get(ptype);
+    if (!ast) {
+      return false;
+    }
+
+    const index = ast.policy.findIndex((r) => util.arrayEquals(r, oldRule));
+    if (index === -1) {
+      return false;
+    }
+
+    const priorityIndex = ast.tokens.indexOf('p_priority');
+
+    if (priorityIndex !== -1) {
+      if (oldRule[priorityIndex] === newRule[priorityIndex]) {
         ast.policy[index] = newRule;
-        return true;
+      } else {
+        this.removePolicy(sec, ptype, oldRule);
+        this.addPolicy(sec, ptype, newRule);
```

Comment:
We can consider throwing an exception here.

Reply:
@hsluoyz any idea?

Reply:
@Zxilly we can first throw exception for now. Let's see how our users feedback.

Reply:
@hsluoyz implement in https://github.com/casbin/node-casbin/pull/250/commits/c4eda9c2535b42e4c9b4b43c53744152f6b5fb30